### PR TITLE
hcl2_upgrade escaped quotes fix

### PR DESCRIPTION
**Pull request by SwampDragons**
Fix quoting issue.

### CODE REVIEW

File: command/test-fixtures/hcl2_upgrade/escaping/input.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+	"variables": {
+		"conf": "{{ env \"ONE\" }}-{{env \"ANOTHER\"}}-{{ env  `BACKTICKED` }}",
+		"nospaces": "{{env \"SOMETHING\"}}",
+		"manyspaces": "{{   env \"ASDFASDF\"}}"
```
Comment (sylviamoss) [inline command/test-fixtures/hcl2_upgrade/escaping/input.json:5]: Nice!

File: command/hcl2_upgrade.go [hunk 2]
```diff
@@ -437,7 +439,22 @@ func transposeTemplatingCalls(s []byte) []byte {
 		Parse(string(s))
 
 	if err != nil {
-		return fallbackReturn(err, s)
+		if strings.Contains(err.Error(), "unexpected \"\\\\\" in operand") {
```
Comment (sylviamoss) [inline command/hcl2_upgrade.go:442]: Nice! I like that this is only done when the parsing fails! 👏🏼

File: command/hcl2_upgrade.go [hunk 3]
```diff
@@ -1247,3 +1279,23 @@ var PASSTHROUGHS = map[string]string{"NVME_Present": "{{ .NVME_Present }}",
 	"ProviderVagrantfile":        "{{ .ProviderVagrantfile }}",
 	"Sound_Present":              "{{ .Sound_Present }}",
 }
+
+func fixQuoting(old string) string {
```
Comment (sylviamoss) [inline command/hcl2_upgrade.go:1283]: 🎉  nice one

**Review by sylviamoss — APPROVED** [review]
Nice - LGTM! 👍🏼

File: command/hcl2_upgrade.go [hunk 4]
```diff
@@ -1247,3 +1279,23 @@ var PASSTHROUGHS = map[string]string{"NVME_Present": "{{ .NVME_Present }}",
...
+	// This regex captures golang template functions that use escaped quotes:
+	// {{ env \"myvar\" }}
+	re := regexp.MustCompile(`{{\s*\w*\s*(\\".*\\")\s*}}`)
```
Comment (azr) [inline command/hcl2_upgrade.go:1286]: Magic ! ✨

**Review by azr — APPROVED** [review]
Good job ! This makes the upgrader way better !

File: command/hcl2_upgrade.go (hunk 2 above)
Comment (azr) [reply command/hcl2_upgrade.go:442]: +1 !

**Review by nywilken — COMMENTED** [review]
Late to the party, but the changes look great 👍

**Pull request merged**